airflow/providers/amazon/aws/sensors/bedrock.py

```diff
@@ -211,3 +262,149 @@ def execute(self, context: Context) -> Any:
+    """
+    Poll the Knowledge Base status until it reaches a terminal state; fails if creation fails.
+
+    .. seealso::
+        For more information on how to use this sensor, take a look at the guide:
+        :ref:`howto/sensor:BedrockKnowledgeBaseActiveSensor`
+
+    :param knowledge_base_id: The unique identifier of the knowledge base for which to get information. (templated)
+
+    :param deferrable: If True, the sensor will operate in deferrable more. This mode requires aiobotocore
+        module to be installed.
+        (default: False, but can be overridden in config file by setting default_deferrable to True)
+    :param poke_interval: Polling period in seconds to check for the status of the job. (default: 5)
+    :param max_retries: Number of times before returning the current state (default: 24)
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is ``None`` or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then default boto3 configuration would be used (and must be
+        maintained on each worker node).
+    :param region_name: AWS region_name. If not specified then the default boto3 behaviour is used.
+    :param verify: Whether or not to verify SSL certificates. See:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html
+    :param botocore_config: Configuration dictionary (key-values) for botocore client. See:
+        https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html
+    """
+
+    INTERMEDIATE_STATES: tuple[str, ...] = ("CREATING", "UPDATING")
+    FAILURE_STATES: tuple[str, ...] = ("DELETING", "FAILED")
+    SUCCESS_STATES: tuple[str, ...] = ("ACTIVE",)
+    FAILURE_MESSAGE = "Bedrock Knowledge Base Active sensor failed."
+
+    template_fields: Sequence[str] = aws_template_fields("knowledge_base_id")
+
+    def __init__(
+        self,
+        *,
+        knowledge_base_id: str,
+        poke_interval: int = 5,
+        max_retries: int = 24,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.poke_interval = poke_interval
+        self.max_retries = max_retries
+        self.knowledge_base_id = knowledge_base_id
+
+    def get_state(self) -> str:
+        return self.hook.conn.get_knowledge_base(knowledgeBaseId=self.knowledge_base_id)["knowledgeBase"][
+            "status"
+        ]
+
+    def execute(self, context: Context) -> Any:
+        if self.deferrable:
+            self.defer(
+                trigger=BedrockKnowledgeBaseActiveTrigger(
+                    knowledge_base_id=self.knowledge_base_id,
+                    waiter_delay=int(self.poke_interval),
+                    waiter_max_attempts=self.max_retries,
+                    aws_conn_id=self.aws_conn_id,
+                ),
+                method_name="poke",
```

Comment:
Why is the callback being set to `poke`? Shouldn't it be set to some method that evaluates whatever the Trigger returns (e.g. the `event` object)?

Reply:
I think this makes sense.   I could write another method that will check the event state, but poke is already doing that.  It results in one extra API call (the extra poke) but makes the code much nicer.   If you look at the execute_complete methods int he other sensors, all they do is `print state and if state != "success" then raise an exception`, that's just a copy/paste subset of the existing poke method.

Reply:
if I modify poke from

```
  def poke(self, context: Context, **kwargs) -> bool:
        state = self.get_state()
        if state in self.FAILURE_STATES:
            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
            if self.soft_fail:
                raise AirflowSkipException(self.FAILURE_MESSAGE)
            raise AirflowException(self.FAILURE_MESSAGE)

        return state not in self.INTERMEDIATE_STATES
 ```
 
 to (untested)
 
 
 ```
   def poke(self, context: Context, **kwargs) -> bool:
        state = kwargs.gget("event").get("state") or self.get_state()
        if state in self.FAILURE_STATES or state == "failure":
            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
            if self.soft_fail:
                raise AirflowSkipException(self.FAILURE_MESSAGE)
            raise AirflowException(self.FAILURE_MESSAGE)

        return state not in self.INTERMEDIATE_STATES
 ```       
 
 Then that would be the same thing without the extra get_state() call

Reply:
That makes sense to me. I think the extra call is worth the clarity. One thing I noticed is that `poke` doesn't accept `event` as a parameter. I'm not sure how important it is the the functionality (I'm sure you've tested it), but the docs seem to strongly recommend it:
>When your operator resumes, Airflow adds a context object and an event object to the kwargs passed to the method_name method. This event object contains the payload from the trigger event that resumed your operator. Depending on the trigger, this can be useful to your operator, like it’s a status code or URL to fetch results. Or, it might be unimportant information, like a datetime. Your method_name method, however, _must_ accept context and event as a keyword argument.

I don't feel too strongly about it either way. 
LGTM

Reply:
Yeah, I initially added it as an explicit parameter but it raised issues with the static checks for not matching the signature of super().  Adding `**kwargs` passed the tests and `event` will just get swallowed up in there.  Since I'm not actually using it, I thought it was acceptable and in the end I decided it actually looked less confusing this way (as opposed to seeing 'event" there and wondering what you are supposed to give it if you aren't familiar with deferrables)


Are you alright with it how it is?  Maybe there's another way around that?  I can look if you care.

Reply:
I'm good with it as is. Just wanted to make sure it was a conscious decision as opposed to an oversight.

Reply:
I don't see the option to resolve the conversation...